### PR TITLE
Add exporting of spots to ellipse ROIs

### PR DESCRIPTION
**Pull request by stelfrich**
This PR adds functionality to export `Spot`s to ellipse ROIs to the ROI Manager. ROIs are not assigned to any channel if the input was a hyperstack.

### CODE REVIEW

File: src/main/java/fiji/plugin/trackmate/extra/action/EllipseRoiExporter.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			double positionX = spot.getFeature( Spot.POSITION_X ) + 0.5;
+			double positionY = spot.getFeature( Spot.POSITION_Y ) + 0.5;
+			double positionZ = spot.getFeature( Spot.POSITION_Z );
+			double frame = spot.getFeature( Spot.FRAME ) + 1;
+
+			double x1 = positionX - radius;
+			double x2 = positionX + radius;
+			double y1 = positionY;
+			double y2 = positionY;
+			Roi roi = new EllipseRoi( x1, y1, x2, y2, 1d ) ;
```
Comment (tinevez) [inline src/main/java/fiji/plugin/trackmate/extra/action/EllipseRoiExporter.java]: This does not seem to take into account the possible physical calibration of the image. 
In the `Spot` object, the coordinates are stored in physical coordinates, but the ImageJ ROIs need to be specified in pixel coordinates. You need to convert them from one coordinate system to another one, using the `dx` value you extract above, and `dy`.
Same for the `z` coordinate.

**Comment by stelfrich** [comment]
@tinevez I fixed the handling of calibrated images!

**Comment by stelfrich** [comment]
Also fixed a weird issue with adding ROIs to the ROI Manager (see http://forum.imagej.net/t/incorrect-positioning-of-roi-when-adding-to-roi-manger-via-script/9647) thanks to @lacan!

**Review by tinevez — CHANGES_REQUESTED** [review]
The issue with setting the position of the ROIs is super weird. I think we met something similar with a project called Filopodyan during last NEUBIAS. I resorted on using 
``` java
int index = imp.getStackIndex(c, z, t);
roi.setPosition( index );
```
instead of 
``` java
roi.setPosition(c,z,t);
```
directly. It worked at the time.

Can I ask you for a favor? Actually two? Or three?
- [ ] Check if my hack works for you as well.
- [ ] Should it be the case, could you invite Wayne Rasband about the issue in a post in the forum?
- [ ] I forgot about 3rd.

**Comment by stelfrich** [comment]
> Check if my hack works for you as well.

The current position of the active `ImagePlus` also overrides a position set with `roi.setPosition(pos)`. So I guess the hack doesn't really work, if that's the issue you wanted to solve. The root cause for this is how `Roi`s are added to the `RoiManager`. If you want to add them to an `Overlay`, `roi.setPosition(...)` works perfectly fine.

How should we proceed?

**Comment by tinevez** [comment]
Arrrh it's not so cool. 
I would like to have a way to create ROIs properly set to the right t,z,c position without relying on an ImagePlus.
Any chance you could contact Wayne about this?

**Comment by stelfrich** [comment]
> I would like to have a way to create ROIs properly set to the right t,z,c position without relying on an ImagePlus.

I have investigated a little more. There is a method `roiManager.add( ImagePlus, Roi, int );` that does not set the position according to the active `ImagePlus` and uses the `Roi`s position instead. The issue here, however, is the last argument `n` that is used to determine the name of the `Roi` when added to the `RoiManager`. You can do `roiManager.add( null, roi, -1 );` which will generate names based on the Roi's x and y position ("YYYY-XXXX"). That, however, are usually not the names that you get with a (hyper)stack because the names are prefixed with the stack position ("PPPP-YYYY-XXXX"). In order to compute a stack position you need the `ImagePlus` for its dimensions...

**TL;DR**: I could use `roiManager.add( null, roi, -1 );` but that will generate names of the form "YYYY-XXXX". What do you think?
